wren-ai-service/src/web/v1/services/sql_explanation.py

```diff
@@ -0,0 +1,123 @@
+import asyncio
+import logging
+from typing import Dict, List, Literal, Optional
+
+from haystack import Pipeline
+from pydantic import BaseModel
+
+from src.utils import async_timer
+
+logger = logging.getLogger("wren-ai-service")
+
+
+# POST /v1/sql-explanations
+class StepWithAnalysisResult(BaseModel):
+    sql: str
+    summary: str
+    sql_analysis_results: List[Dict]
+
+
+class SQLExplanationRequest(BaseModel):
+    _query_id: str | None = None
+    question: str
+    steps_with_analysis_results: List[StepWithAnalysisResult]
+
+    @property
+    def query_id(self) -> str:
+        return self._query_id
+
+    @query_id.setter
+    def query_id(self, query_id: str):
+        self._query_id = query_id
+
+
+class SQLExplanationResponse(BaseModel):
+    query_id: str
+
+
+# GET /v1/sql-explanations/{query_id}/result
+class SQLExplanationResultRequest(BaseModel):
+    query_id: str
+
+
+class SQLExplanationResultResponse(BaseModel):
+    class SQLExplanationResultError(BaseModel):
+        code: Literal["OTHERS"]
+        message: str
+
+    status: Literal["understanding", "generating", "finished", "failed"]
+    response: Optional[List[List[Dict]]] = None
+    error: Optional[SQLExplanationResultError] = None
+
+
+class SQLExplanationService:
+    def __init__(self, pipelines: dict[str, Pipeline]):
+        self._pipelines = pipelines
+        self.sql_explanation_results: dict[str, SQLExplanationResultResponse] = {}
+
+    @async_timer
+    async def sql_explanation(self, sql_explanation_request: SQLExplanationRequest):
+        try:
+            query_id = sql_explanation_request.query_id
+
+            self.sql_explanation_results[query_id] = SQLExplanationResultResponse(
+                status="understanding",
+            )
+
+            self.sql_explanation_results[query_id] = SQLExplanationResultResponse(
+                status="generating",
+            )
+
+            sql_explanation_results = []
```

Comment:
It seems that it hasn’t worked in the workflow and is only present in the debug log. I think we could remove it and also modify or remove the debug log.